src/NServiceBus.Core/Hosting/StartupDiagnostics/HostStartupDiagnostics.cs

```diff
@@ -45,12 +45,23 @@ static WriteStartupDiagnostics GetDiagnosticsWriter(ReadOnlySettings settings)
                 catch (Exception e)
                 {
                     logger.Error("Unable to determine the diagnostics output directory. Check the attached exception for further information, or configure a custom diagnostics directory using 'EndpointConfiguration.SetDiagnosticsPath()'.", e);
+
+                    return data => TaskEx.CompletedTask;
                 }
             }
 
             if (!Directory.Exists(diagnosticsRootPath))
             {
-                Directory.CreateDirectory(diagnosticsRootPath);
+                try
+                {
+                    Directory.CreateDirectory(diagnosticsRootPath);
+                }
+                catch(Exception e)
+                {
+                    logger.Error("Unable to create the diagnostics output directory. Check the attached exception for further information, or configure a custom diagnostics directory using 'EndpointConfiguration.SetDiagnosticsPath()'.", e);
```

Comment:
The text of the error message might need to be tweaked a bit here since you could hit this when `SetDiagnosticsPath` as already been called with path that can't be created. Not sure that's worth trying to distinguish though.

Reply:
the message is good enough to start with imo. Maybe we could change the "or configure a custom diagnostics directory" to "or change the diagnostics directory".

Reply:
Should it be enhanced with something like:
```suggestion
                    logger.Error("Unable to create the diagnostics output directory. Check the attached exception for further information, or configure a custom diagnostics directory using 'EndpointConfiguration.SetDiagnosticsPath()'. In case a custom diagnostic path is already configured ensure the account running the application has write permissions to the configured path.", e);
```